fpsim/empowerment.py

```diff
@@ -0,0 +1,289 @@
+    partnered, partnership_age = get_partnership_init_vals(ppl)
+
+    # Populate states
+    ppl.partnered = partnered
+    ppl.partnership_age = partnership_age
+
+
+def init_empowerment_states(ppl):
+    """
+    If ppl.pars['use_empowerment'] == True, location-specific data
+    are expected to exist to populate empowerment states/attributes.
+
+    If If ppl.pars['use_empowerment'] == False, related
+    attributes will be initialized with values found in defaults.py.
+    """
+
+    # Initialize empowerment-related attributes with location-specific data
+    # TODO: check whether the corresponding data dictionary exists in ppl.pars,
+    # if it doesn't these states could be intialised with a user-defined distribution.
+    emp = get_empowerment_init_vals(ppl)
+
+    # Populate empowerment states with location-specific data
+    ppl.paid_employment = emp['paid_employment']
+    ppl.sexual_autonomy = emp['sexual_autonomy']
+    ppl.decision_wages  = emp['decision_wages']   # Decision-making autonomy over household purchases/wages
+    ppl.decision_health = emp['decision_health']  # Decision-making autonomy over her health
+    return
+
+
+def init_education_states(ppl):
+    """
+    If ppl.pars['use_education'] == True, location-specific data
+    are expected to exist to populate education states/attributes.
+
+    If If ppl.pars['use_education'] == False, related
+    attributes will be initialized with values found in defaults.py.
+    """
+
+    # Check that people have requested to use urban/rural information
+    if not ppl.pars['use_urban']:
+        mssg = f"To use the education module, you need to set use pars['use_urban'] = True (current value: {ppl.pars['use_urban']}."
+        raise ValueError(mssg)
+
+    # Get distributions of initial values from location-specific data
+    edu = get_education_init_vals(ppl)
+
+    # Populate People's education states
+    ppl.edu_objective = edu['edu_objective']  # Highest-ideal level of education to be completed (in years)
+    ppl.edu_attainment = edu['edu_attainment']  # Current level of education achieved in years
+    ppl.edu_dropout = edu['edu_dropout']  # Whether a person has dropped out before reaching their goal
+    ppl.edu_interrupted = edu['edu_interrupted']  # Whether a person/woman has had their education temporarily interrupted, but can resume
+    ppl.edu_completed = edu['edu_completed']  # Whether a person/woman has reached their education goals
+    ppl.edu_started = edu['edu_started']  # Whether a person/woman has started thier education
+
+
+def get_urban_init_vals(ppl, urban_prop=None):
+    """ Get initial distribution of urban """
+
+    n = len(ppl)
+    urban = np.ones(n, dtype=bool)
+
+    if urban_prop is None:
+        if ppl.pars['urban_prop'] is not None:
+            urban_prop = ppl.pars['urban_prop']
+
+    if urban_prop is not None:
+        urban = fpu.n_binomial(urban_prop, n)
+
+    return urban
+
+
+def get_partnership_init_vals(ppl):
```

Comment:
Renamed `initialize_attribute` --> `get_attr_init_vals()` which i don't like very much but couldn't come up with a better name (get_attr_vals_from_data?). At present, these methods make use of location-specific data that are expected to exist in `ppl.pars`, to set the initial conditions of the corresponding people attributes (states). 

Some of these `get_ ...` methods could also be used to update the attributes during a simulation. eg, if there isn't an `update()` method to say how probability of having paid employment evolves over time, we could update this probability based on the age of the agent because we have this data in pars, stratified by age.